File: java/Shubham Sigdar/DSA/Array/GoalParserInterpretation.java
package DSA.Array;
//You own a Goal Parser that can interpret a string command. The command consists of an alphabet of "G", "()" and/or "(al)" in some order. The Goal Parser will interpret "G" as the string "G", "()" as the string "o", and "(al)" as the string "al". The interpreted strings are then concatenated in the original order.
//
//        Given the string command, return the Goal Parser's interpretation of command.
//
//
//
//        Example 1:
//
//        Input: command = "G()(al)"
//        Output: "Goal"
//        Explanation: The Goal Parser interprets the command as follows:
//        G -> G
//        () -> o
//        (al) -> al
//        The final concatenated result is "Goal".
//        Example 2:
//
//        Input: command = "G()()()()(al)"
//        Output: "Gooooal"
//        Example 3:
//
//        Input: command = "(al)G(al)()()G"
//        Output: "alGalooG"
public class GoalParserInterpretation {
    public String interpret(String command) {
        StringBuilder s = new StringBuilder(command.length());
        for(int i=0; i<command.length();i++){
            if(command.charAt(i)=='G'){
                s.append("G");
            }else if(command.charAt(i)=='('){
                if(command.charAt(i+1)==')'){
                    s.append("o");
                }else if(command.charAt(i+1)=='a'){
                    s.append("al");
                }
            }
        }
        return s.toString();
    }
}

File: javascript/Shubham Sigdar/leetcode/Goal Parser Interpretation.js
// Example 1:

// Input: command = "G()(al)"
// Output: "Goal"
// Explanation: The Goal Parser interprets the command as follows:
// G -> G
// () -> o
// (al) -> al
// The final concatenated result is "Goal".
// Example 2:

// Input: command = "G()()()()(al)"
// Output: "Gooooal"
// Example 3:

// Input: command = "(al)G(al)()()G"
// Output: "alGalooG"
/**
 * @param {string} command
 * @return {string}
 */

//P: always string, no empty string
//R: string
//E: G()()()()(al)=>Gooooal
var interpret = function(command) {
    let str=[];
   for(let i=0;i<command.length;i++){
     if(command[i]=="G")
       str.push("G");
     else if(command[i]=="(" && command[i+1]==")"){
      str.push("o");
      i+=1
     } 
     else if(command[i]=="(" && command[i+1]=="a"){
      str.push("al");
       i+=3;
     }
       
   }
   return str.join("");
  };
  // var interpret = function(command) {
  //   console.log(command.split("()").join("o").split("(al)").join("al"))
  // };
  // interpret("G()()(al)")